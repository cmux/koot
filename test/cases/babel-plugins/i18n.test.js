const path = require('path');
const babel = require('@babel/core');
const plugin = require('../../../packages/koot-webpack/loaders/babel/plugins/i18n');

const examples = [
    `__('test')`,
    `__('object')`,
    `__('object.value1')`,
    `__('object', 'value1')`,
    `__('object.value_with_parameter')`,
    `__('object', 'value_with_parameter')`,
    `__('object.value_with_parameter', {insert: 'TEST'})`,
    `__('object', 'value_with_parameter', {insert: 'TEST'})`,

    `const t = "value1"; __('object', t)`,

    `let t;
__('object.value_with_parameter', {
    insert: t || undefined
})`,

    `let t;
__('object.value_with_parameter', {
insert: t || 'undefined'
})`,

    `let t;
__('object.value_with_parameter', {
insert: t || undefined
});
__('object', 'value_with_parameter', {insert: 'TEST'})`,
];

describe('Babel Plugin: i18n', () => {
    examples.forEach((codeOriginal, index) => {
        it(`${index}`, () => {
            const { code } = babel.transform(codeOriginal, {
                plugins: [
                    [
                        plugin,
                        {
                            stage: 'client',
                            functionName: '__',
                            localeId: 'zh',
                            localeFile: path.resolve(
                                __dirname,
                                './i18n.locales.json'
                            ),
                        },
                    ],
                ],
            });
            expect(code).toMatchSnapshot();
        });
    });
});
