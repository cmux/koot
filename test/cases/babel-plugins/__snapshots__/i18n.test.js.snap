// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel Plugin: i18n 0 1`] = `"\\"TEST\\";"`;

exports[`Babel Plugin: i18n 1 1`] = `
"({
  \\"value1\\": \\"O_VALUE_1\\",
  \\"value_with_parameter\\": \\"O_VALUE_PARA_\${insert}\\",
  \\"nest\\": {
    \\"value1\\": \\"O_NEST_VALUE_1\\"
  }
});"
`;

exports[`Babel Plugin: i18n 2 1`] = `"\\"O_VALUE_1\\";"`;

exports[`Babel Plugin: i18n 3 1`] = `
"import __ from \\"koot/i18n/translate\\";

__({
  \\"value1\\": \\"O_VALUE_1\\",
  \\"value_with_parameter\\": \\"O_VALUE_PARA_\${insert}\\",
  \\"nest\\": {
    \\"value1\\": \\"O_NEST_VALUE_1\\"
  }
}, \\"value1\\");"
`;

exports[`Babel Plugin: i18n 4 1`] = `"\\"O_VALUE_PARA_\${insert}\\";"`;

exports[`Babel Plugin: i18n 5 1`] = `
"import __ from \\"koot/i18n/translate\\";

__({
  \\"value1\\": \\"O_VALUE_1\\",
  \\"value_with_parameter\\": \\"O_VALUE_PARA_\${insert}\\",
  \\"nest\\": {
    \\"value1\\": \\"O_NEST_VALUE_1\\"
  }
}, \\"value_with_parameter\\");"
`;

exports[`Babel Plugin: i18n 6 1`] = `
"import __ from \\"koot/i18n/translate\\";

__(\\"O_VALUE_PARA_\${insert}\\", {
  insert: 'TEST'
});"
`;

exports[`Babel Plugin: i18n 7 1`] = `
"import __ from \\"koot/i18n/translate\\";

__({
  \\"value1\\": \\"O_VALUE_1\\",
  \\"value_with_parameter\\": \\"O_VALUE_PARA_\${insert}\\",
  \\"nest\\": {
    \\"value1\\": \\"O_NEST_VALUE_1\\"
  }
}, \\"value_with_parameter\\", {
  insert: 'TEST'
});"
`;

exports[`Babel Plugin: i18n 8 1`] = `
"import __ from \\"koot/i18n/translate\\";

__({
  \\"value1\\": \\"O_VALUE_1\\",
  \\"value_with_parameter\\": \\"O_VALUE_PARA_\${insert}\\",
  \\"nest\\": {
    \\"value1\\": \\"O_NEST_VALUE_1\\"
  }
}, \\"value_with_parameter\\", {
  insert: 'TEST'
});"
`;

exports[`Babel Plugin: i18n 9 1`] = `
"import __ from \\"koot/i18n/translate\\";
const t = \\"value1\\";

__({
  \\"value1\\": \\"O_VALUE_1\\",
  \\"value_with_parameter\\": \\"O_VALUE_PARA_\${insert}\\",
  \\"nest\\": {
    \\"value1\\": \\"O_NEST_VALUE_1\\"
  }
}, t);"
`;

exports[`Babel Plugin: i18n 10 1`] = `
"import __ from \\"koot/i18n/translate\\";
let t;

__(\\"O_VALUE_PARA_\${insert}\\", {
  insert: t || undefined
});"
`;

exports[`Babel Plugin: i18n 11 1`] = `
"import __ from \\"koot/i18n/translate\\";
let t;

__(\\"O_VALUE_PARA_\${insert}\\", {
  insert: t || 'undefined'
});"
`;

exports[`Babel Plugin: i18n 12 1`] = `
"import __ from \\"koot/i18n/translate\\";
let t;

__(\\"O_VALUE_PARA_\${insert}\\", {
  insert: t || undefined
});

__({
  \\"value1\\": \\"O_VALUE_1\\",
  \\"value_with_parameter\\": \\"O_VALUE_PARA_\${insert}\\",
  \\"nest\\": {
    \\"value1\\": \\"O_NEST_VALUE_1\\"
  }
}, \\"value_with_parameter\\", {
  insert: 'TEST'
});"
`;
